ReportingServicesTools/Functions/CatalogItems/Rest/Write-RsRestCatalogItem.ps1

```diff
@@ -239,24 +311,46 @@ function Write-RsRestCatalogItem
+                {
+                    Write-Verbose "Uploading $EntirePath to $RsFolder..."
 
-                $payloadJson = ConvertTo-Json $payload
+                    $payloadJson = ConvertTo-Json $payload
 
-                if ($Credential -ne $null)
-                {
-                    Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -Credential $Credential -Verbose:$false | Out-Null
+                    if ($Credential -ne $null)
+                    {
+                        Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -Credential $Credential -Verbose:$false | Out-Null
+                    }
+                    else
+                    {
+                        Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -UseDefaultCredentials -Verbose:$false | Out-Null
+                    }
                 }
-                else
+
+                if ($itemType -eq "PowerBIReport" -and $isLargePowerBIReport -eq $true)
                 {
-                    Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -UseDefaultCredentials -Verbose:$false | Out-Null
+                    Write-Verbose "Uploading $EntirePath to $RsFolder via endpoint for large files..."
+                    
+                    $headers = @{
+                        "accept"="application/json, text/plain, */*"
+                    }
+                    $uri = [String]::Format($powerBIReportsByPathApi, $itemPath)
+
+                    if ($Credential -ne $null)
+                    {
+                        Invoke-RestMethod -Uri $uri -Method Post -WebSession $WebSession -Body $bodyLines -ContentType "multipart/form-data; boundary=$boundary" -Headers $headers -Credential $Credential -Verbose:$false | Out-Null
+                    }
+                    else
+                    {
+                        Invoke-RestMethod -Uri $uri -Method Post -WebSession $WebSession -Body $bodyLines -ContentType "multipart/form-data; boundary=$boundary" -Headers $headers -UseDefaultCredentials -Verbose:$false | Out-Null
+                    }
                 }
 
                 Write-Verbose "$EntirePath was uploaded to $RsFolder successfully!"
             }
             catch
             {
-                if ($_.Exception.Response -ne $null -and $_.Exception.Response.StatusCode -eq 409 -and $Overwrite)
+                if ($isLargePowerBIReport -ne $true -and $_.Exception.Response -ne $null -and $_.Exception.Response.StatusCode -eq 409 -and $Overwrite)
```

Comment:
why the code below doesn't apply for large pbix, what will happen if there is an existing item?

Reply:
If we use endpoint for a large PBIX report, we do not receive a 409 code from the server, because we always rewrite this catalog item